packages/components/src/internal/components/user/ProfilePage.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
+ * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
+ */
+import React, { FC, useState } from 'react';
+import { Button } from 'react-bootstrap';
+
+import { createNotification, withTimeout } from '../notifications/actions';
+import { queryGridInvalidate } from '../../actions';
+import { SCHEMAS } from '../../schemas';
+import { isLoginAutoRedirectEnabled } from '../administration/utils';
+import { InsufficientPermissionsPage } from '../permissions/InsufficientPermissionsPage';
+
+import { Page } from '../base/Page';
+
+import { Section } from '../base/Section';
+
+import { Notification } from '../notifications/Notification';
+
+import { useServerContext } from '../base/ServerContext';
+
+import { App } from '../../../index';
+
+import { UserDetailHeader } from './UserDetailHeader';
+import { getUserRoleDisplay } from './actions';
+
+import { UserProfile } from './UserProfile';
+import { ChangePasswordModal } from './ChangePasswordModal';
+
+import { useUserProperties } from './UserProvider';
+
+interface Props {
+    updateUserDisplayName: (displayName: string) => any;
+    setReloadRequired: () => any;
+    goBack: (n?: number) => any;
```

Comment:
Came back to this and after some memory-wracking, I don't quite recall why we wanted to add in withRouteLeave? It doesn't seem to be using any props from it, and instead gets all of the props it needs from the dispatch—I remember talking about wanting a subset of the product-side CommonDispatchProps, but not wanting to relocate that to ui-components, but not how that factored into the solution gestured at by my comments.
Anyway, take a look and let me know what you think!

Reply:
What you have is good.